messagebus/src/vespa/messagebus/network/rpctargetpool.cpp

```diff
@@ -4,23 +4,41 @@
 
 namespace mbus {
 
-RPCTargetPool::Entry::Entry(RPCTarget::SP target, uint64_t lastUse) :
-    _target(target),
-    _lastUse(lastUse)
+RPCTargetPool::Entry::Entry(std::vector<RPCTarget::SP> targets, uint64_t lastUse)
+    : _targets(std::move(targets)),
+      _lastUse(lastUse),
+      _next(0)
 { }
 
-RPCTargetPool::RPCTargetPool(double expireSecs) :
-    _lock(),
-    _targets(),
-    _timer(new SteadyTimer()),
-    _expireMillis(static_cast<uint64_t>(expireSecs * 1000))
+RPCTarget::SP
+RPCTargetPool::Entry::getTarget(const LockGuard &, uint64_t now) {
+    if (_next >= _targets.size()) {
+        _next = 0;
+    }
+    RPCTarget::SP target =  _targets[_next++];
+    if ( ! target->isValid()) return RPCTarget::SP();
+    _lastUse = now;
+    return target;
+}
+
+bool
+RPCTargetPool::Entry::inUse(const LockGuard &) const {
+    for (const auto & target : _targets) {
+        if (target.use_count() > 1) return true;
```

Comment:
Should this be "_> 0_" instead ?

Reply:
Consider using braces.

Reply:
Braces done.
target is a shared_ptr so use_count will never be 0. If it is larger than 1 it means there is another reference than the one in the pool active.